utils/kubernetes/service.go

```diff
@@ -73,5 +75,20 @@ func GetEndpoint(ctx context.Context, opts *ServiceOptions, obj *corev1.Service)
 		}
 	}
 
+	if !utils.TcpCheck(endpoint.External) {
+		if len(strings.SplitAfter(opts.APIServerURL, "://")) > 1 {
+			endpoint.External.Address = strings.SplitAfter(strings.SplitAfter(opts.APIServerURL, "://")[1], ":")[0]
+			if len(endpoint.External.Address) > 0 {
+				endpoint.External.Address = endpoint.External.Address[:len(endpoint.External.Address)-1]
+			}
+		}
+		if !utils.TcpCheck(endpoint.External) {
+			endpoint.External.Port = nodePort
```

Comment:
... nope, try the `NodePort`

Reply:
U mean `port.NodePort`? If yes, this is exactly the same as that.

Reply:
Ah yes.